File: 50dias/cpp/24.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;

int GeraNumeroEntre1e10() {
    // rand() % 10 gera um numero entre 0 e 9
    return (rand() % 10) + 1;
}

bool EfetuaQuestao(int multiplicando, int multiplicador) {
    int resposta;
    cout << multiplicando << "x" << multiplicador << " = ";
    cin >> resposta;

    return resposta == multiplicador * multiplicando;
}

void UsuarioErrou(int multiplicando, int multiplicador) {
    cout << "Infelizmente voce errou!" << endl;
    cout << "O resultado de " << multiplicando << "x";
    cout << multiplicador << " = " << multiplicando * multiplicador;
    cout << endl << endl;
}

bool UsuarioQuerContinuar() {
    char tentarNovamente;
    cout << "Voce quer tentar novamente? [s]im/[n]ao : ";
    cin >> tentarNovamente;
    return tentarNovamente == 's';
}

int main()
{
    srand(time(0));
    bool continuar = true;
    int multiplicando = 0;
    int multiplicador = 0;
    do {
        while (continuar) {
            multiplicando = GeraNumeroEntre1e10();
            multiplicador = GeraNumeroEntre1e10();

            continuar = EfetuaQuestao(multiplicando,
                multiplicador);
        }

        UsuarioErrou(multiplicando, multiplicador);
        continuar = UsuarioQuerContinuar();
    } while (continuar);

    return 0;
}

File: 50dias/python/24.py
import random

def GeraNumeroEntre1e10():
    return random.randint(1,10)

def EfetuaQuestao(multiplicando, multiplicador):
    resposta = int(input(str(multiplicando) + 'x'+str(multiplicador)+' = '))
    return resposta == multiplicando*multiplicador

def UsuarioErrou(multiplicando, multiplicador):
    print('Infelizmente voce errou!\n')
    print('O resultado de {prim}x{sec} = {ter}'.format(prim=multiplicando,
                                                       sec=multiplicador,
                                                       ter=multiplicando*multiplicador))
    print('\n\n')

def UsuarioQuerContinuar():
    tentar = input('Voce quer tentar novamente? [s]im/[n]ao: ')
    return tentar == 's'

def main():
    continuar = True

    while continuar:
        while continuar:
            multiplicando = GeraNumeroEntre1e10()
            multiplicador = GeraNumeroEntre1e10()

            continuar = EfetuaQuestao(multiplicando, multiplicador)

        UsuarioErrou(multiplicando, multiplicador)
        continuar = UsuarioQuerContinuar()


main()